pkg/shell/superuser.jsx

```diff
@@ -73,6 +73,11 @@ class UnlockDialog extends React.Component {
                                 value={state.prompt.value}
                             />
                         </FormGroup>
+                        <FormGroup>
+                            <Checkbox onChange={this.props.change_remember_access_level_on_unlock}
+                                      isChecked={this.props.remember_access_level_on_unlock}
+                                      label={ _("Automatically switch to administrative access on next login") } />
```

Comment:
This may not always work, right? Especially when you use SSH keys (which we heavily promote now) or 2FA. So ISTM that this promises too much for SSH. Can we restrict this to localhost only, and always make it explicit for SSH hosts?

Reply:
Could you elaborate?  Should we have no checkbox at all for connections via SSH?

Reply:
I haven't thought this through, and I don't have a consistent proposal yet -- I just think that like this it's too confusing. It sounds like Cockpit would do long-term credential storage somehow, or that this is a promise. In many cases it's not technically possible to "automatically switch to admin access". It will sometimes fail locally as well, but much more often with SSH.

Maybe a wording change like "Automatically attempt this on next login" would suffice already. As it's off by default, admins then at least have a choice.

I could also live with not showing the checkbox at all for remote hosts (and never attempting sudo automatically, ever). However, I think we should keep that possibility for localhost at least, both for convenience and backwards compatibility. This would then be a pretty arbitrary design choice.

Reply:
Right, in that spirit. I was aiming for something shorter, as this is both rather long, and also needs to be translated. But maybe @garrett has a better idea here.

Reply:
I think we can be even more exact (and verbose): "Try to automatically switch to administrative access on next login by providing the login password also to sudo".

Reply:
Uh... if it's just next time someone logs in, then it's kind of pointless, especially if someone has to check that each time.

What about something simpler?

"Use administration mode when logging in."

---

Also, where can someone undo this? Would we require them to drop their admin privs and then try to raise them again? That seems a bit odd, if so. The right place to undo something like, _after being set_ in the admin prompt, this would be under the session menu, as a toggle.

(Simply put: Every "remember" checkbox in a modal needs to have a setting somewhere outside that modal to undo it.)

Reply:
Perhaps even toss in an always:

☑ Always use administration mode when logging in.

Reply:
> Also, where can someone undo this? Would we require them to drop their admin privs and then try to raise them again?

The way it works right now in this PR is that the "Switch to limited access" dialog will do this.  The dialog also explains this: "Your browser will remember your access level across sessions."

And, yeah, this is diffcult for the user to put all this together and understand what is really happening...